sdk/src/Core/Amazon.Runtime/Internal/Util/StringUtils.cs

```diff
@@ -243,6 +243,14 @@ public static string FromList(List<Enum> values)
         /// <returns>Header value representing the list of T</returns>
         public static string FromValueTypeList<T>(List<T> values)  where T : struct
         {
+            // ToString() on boolean types automatically Pascal Cases. Xml-based protocols
+            // are case sensitive and accept "true" and "false" as the valid set of booleans.
+            // and we should be returning the booleans as is.
+            if ((bool)(values?.Any(x => x.GetType() == typeof(bool))))
```

Comment:
More simplified: `if (typeof(T) == typeof(bool))` since all items in values must be of type T

Reply:
good point, updated